parse_jenkins_builds.py

```diff
@@ -169,6 +170,67 @@ def grep(filename, pattern, verbose=False):
     payload["wait_time"] = current_time - queue_time
     payload["start_time"] = 0
 
+    kill_index = 0
+
+    # Abort stuck rocm jobs
+    if (
+        job_name in ("ib-run-pr-unittests", "ib-run-pr-relvals", "ib-run-baseline")
+        and reason.endswith("-offline")
+        and (payload["wait_time"] / 1000 / 60 > 60)
+    ):
+        params = element["params"].strip().split("\n")
+        main_params = ""
+        other_params = []
+        context = ""
+        upload_unique_id = ""
+        pull_request = ""
+        commit = ""
+
+        for _ in params:
+            k, v = _.split("=")
+            if k == "PULL_REQUEST":
+                main_params = _
+                pull_request = v
+            else:
+                if k == "CONTEXT_PREFIX":
+                    context = v
+                if k == "UPLOAD_UNIQ_ID":
+                    upload_unique_id = v
+
+                other_params.append(_)
+
+        if "GPU_FLAVOR=rocm" in other_params or "TEST_FLAVOR=rocm" in other_params:
+            with open("abort-{0}.prop".format(kill_index), "w") as f:
+                f.write("JENKINS_PROJECT_TO_KILL={0}\n".format(job_name))
+                f.write("JENKINS_PROJECT_PARAMS={0}\n".format(main_params))
+                f.write("EXTRA_PARAMS={0}\n".format(";".join(other_params)))
+
+            kill_index += 1
+
+            repository, pr = pull_request.split("#", 1)
+
+            if upload_unique_id:
+                with urllib.request.urlopen(
+                    "http://localhost/SDT/jenkins-artifacts/pull-request-integration/{0}/prs_commits.txt".format(
```

Comment:
@iarspider , why localhost? this job runs on jenkins controller while the prs_commits.txt is available in cmssdt server